posthog/session_recordings/queries/session_replay_events.py

```diff
@@ -189,72 +189,49 @@ def get_events_for_session(self, session_id: str, team: Team, limit: int = 100)
     def get_similar_recordings(
         self, session_id: str, team: Team, limit: int = 10, similarity_range: float = 0.5
     ) -> list[dict]:
-        """Find recordings with similar event sequences.
-        The similarity is based on:
-        1. Having similar event patterns (types and order)
-        2. Similar activity metrics (clicks, keypresses, etc)
-        3. Similar duration
-
+        """Find recordings with similar URL sequences.
         Args:
             session_id: The session ID to find similar recordings for
             team: The team the recording belongs to
             limit: Maximum number of similar recordings to return
-            similarity_range: How similar the recordings should be (0.0 to 1.0). Default 0.5 means within 50% range
+            similarity_range: How similar the recordings should be (0.0 to 1.0)
         """
-        # First get the target recording's metadata
-        target_metadata = self.get_metadata(session_id=session_id, team=team)
-        if not target_metadata:
-            return []
-
-        # Get target recording's events
-        target_events = self.get_events_for_session(session_id, team, limit=100)
-        if not target_events:
-            return []
-
-        # Build event pattern string for comparison
-        target_pattern = ",".join([e["event"] for e in target_events])
-
         query = """
+        WITH target_urls AS (
+            SELECT groupArrayArray(all_urls) as url_sequence
+            FROM session_replay_events
+            WHERE team_id = %(team_id)s
+            AND session_id = %(session_id)s
+        )
         SELECT
             sre.session_id,
             sre.distinct_id,
-            min_first_timestamp,
-            max_last_timestamp,
+            min_first_timestamp as start_time,
+            max_last_timestamp as end_time,
             click_count,
             keypress_count,
             mouse_activity_count,
             active_milliseconds,
-            arrayStringConcat(groupArray(e.event), ',') as event_pattern
+            all_urls as urls,
+            target_urls.url_sequence
         FROM session_replay_events sre
-        JOIN events e ON e.team_id = sre.team_id AND e.$session_id = sre.session_id
+        CROSS JOIN target_urls
```

Comment:
is there a clever reason i'm not seeing for joining on target urls?
i'd worry this is slow (although measurement > worrying :))

since target_urls is fixed I guess we can 

`having groupArrayArray(all_urls) = target_urls` to avoid the cross join

but i'm interested if the cross join is giving us something i'm missing :)